apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/FDADrugUtils.java

```diff
@@ -0,0 +1,152 @@
+package gov.cms.bfd.server.war;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FDADrugUtils implements IDrugCodeProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FDADrugUtils.class);
+  /** Stores the PRODUCTNDC and SUBSTANCENAME from the downloaded NDC file. */
+  private static Map<String, String> ndcProductMap = null;
```

Comment:
This is more a style argument, so not a criticism at all because we see it in many places in BFD. So we have essentially a Singleton pattern here which could be instantiated in any number of ways. Ideally we'd have DI (dependency injection ) but we could also treat it as singleton which could be instantiated on an as needed basis (i.e., a factory method) anywhere in our codebase. The reason I bring it up is that you had to jump through a bunch of hoops passing around this class; granted, via its interface, but it still pollutes method signatures which ends up being passed down the stack any number of times.

Reply:
This is using dependency injection now (see the changes in SpringConfiguration) but I believe the problem is that all of the consuming classes were static so they cannot inject the bean. Open to concrete suggestions on how Chris could make that better in this PR without refactoring the entire thing.

Reply:
I think best answer here is a refactor to make the transformer classes actual instantiated objects that can inject dependencies. This is also a problem with the MetricsRegistry. That I believe it outside the scope of this PR though.